roachtest: pgjdbc failed [DECLARE panics when sent as a prepared statement]

roachtest.pgjdbc [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4427196&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4427196&tab=artifacts#/pgjdbc) on master @ [4c65a10fea9762b97559c1a488cd5b96cb2cf81a](https://github.com/cockroachdb/cockroach/commits/4c65a10fea9762b97559c1a488cd5b96cb2cf81a):


```
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.breakDescribeOnFirstServerPreparedExecution[returningInQuery = AB, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.XmlTest.testDoubleSet - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testGetLargeUpdateCountStatementSELECT[binary = FORCE, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310InfinityTests.test[binary = REGULAR, expr = -infinity, pgType = date, klass = class java.time.LocalDate] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.UUIDTest.testUUID[binary=REGULAR, stringType=UNSPECIFIED] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.optional.SimpleDataSourceWithSetURLTest.testGetConnection - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.optional.PoolingDataSourceTest.testCreateDataSource - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaPreparedExecuteQuery[returningInQuery = AB, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.core.NativeQueryBindLengthTest.testBindLengthCalculation['$1$2...$9$10..$99$100'.length = 2*9+3*90+4 == 292] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310Test.testGetLocalDate[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PGObjectGetTest.getAsString[binary = REGULAR, sql = null::interval, type = class org.postgresql.util.PGInterval] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.SetObject310Test.testSetLocalDateTimeBc[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PGObjectGetTest.getAsPGobjectSubtype[binary = REGULAR, sql = null::inet, type = class org.postgresql.util.PGobject] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testWithInsertSelect[returningInQuery = STAR, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.replication.ReplicationSlotTest.testNotAvailableCreateLogicalSlotWithoutOutputPlugin - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testDelete[returningInQuery = NO, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.SetObject310Test.testSetLocalDateWithoutType[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310InfinityTests.test[binary = FORCE, expr = infinity, pgType = date, klass = class java.time.LocalDate] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.SetObject310Test.testTimeStampRounding[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaPreparedExecuteQuery[returningInQuery = NO, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.BinaryStreamTest.testKnownLength10Kb - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testUpdate[returningInQuery = NO, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.util.ByteStreamWriterTest.testIOExceptionPassedThroughAsCause - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testPSUpdate[returningInQuery = STAR, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testPSUpdate[returningInQuery = STAR, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310Test.testGetLocalTimeNull[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.extensions.HStoreTest.testHStoreSendEscaped - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310Test.testProlepticCalendarTimestamp[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testNullExecuteLargeBatchPreparedStatement[binary = REGULAR, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.BinaryStreamTest.testKnownLength200Kb - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.ConnectionValidTimeoutTest.testIsValidRespectsSmallerTimeout[networkTimeoutMillis=500, validationTimeoutSeconds=1, expectedMaxValidationTimeMillis=600] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testNullExecuteLargeBatchStatement[binary = REGULAR, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.optional.ConnectionPoolTest.testNotPooledConnection - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.Jdbc3SavepointTest.testGlobalRollbackWorks - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaPreparedExecuteQuery[returningInQuery = STAR, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.ClientInfoTest.testExplicitSetAppNameNotificationIsParsed - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.SetObject310Test.testSetOffsetDateTime[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testWithInsertSelect[returningInQuery = NO, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.XmlTest.testDOMRead - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.CharacterStreamTest.testKnownLongLength100Kb - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.WrapperTest.testConnectionIsWrapperForPGConnection - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaPreparedExecuteQuery[returningInQuery = AB, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.SetObject310Test.testSetLocalTimeWithType[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.core.NativeQueryBindLengthTest.testBindLengthCalculation['$1$2...$9'.length = 2*9 == 18] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.optional.SimpleDataSourceWithUrlTest.testDdlOverConnection - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.XmlTest.testFree - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.optional.ConnectionPoolTest.testPreparedStatementConnection - unknown (unexpected)
		For a full summary look at the pgjdbc artifacts 
		
		An updated blocklist (pgjdbcBlockList22_1) is available in the artifacts' pgjdbc log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pgjdbc.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.pgjdbc [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4436786&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4436786&tab=artifacts#/pgjdbc) on master @ [1d13605e2aeb1af0a5536f1f460e45d557119128](https://github.com/cockroachdb/cockroach/commits/1d13605e2aeb1af0a5536f1f460e45d557119128):


```
		--- FAIL: org.postgresql.test.jdbc2.ResultSetTest.testZeroRowResultPositioning - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.NumericTransferTest.sendReceive100000[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[6: batchTest(mode=NO_FAIL_SELECT, position=NONE, autoCommit=NO, batchType=SIMPLE, generateKeys=NO, binary=FORCE, insertRewrite=true)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PreparedStatementTest.testAlternatingBindType[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[126: batchTest(mode=FAIL_VIA_DUP_KEY, position=ALMOST_LAST_ROW, autoCommit=NO, batchType=PREPARED, generateKeys=NO, binary=FORCE, insertRewrite=true)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaPreparedExecute[returningInQuery = NO, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[49: batchTest(mode=FAIL_VIA_SELECT_RUNTIME, position=FIRST_ROW, autoCommit=YES, batchType=SIMPLE, generateKeys=YES, binary=REGULAR, insertRewrite=false)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.CursorFetchTest.testNoRowResultPositioning[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ReplaceProcessingTest.run[input={fn timestampadd(SQL_TSI_WEEK, ?, {fn now()})}, expected=(CAST( $1||' week' as interval)+ now())] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testWithInsertSelect[returningInQuery = STAR, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[110: batchTest(mode=FAIL_VIA_DUP_KEY, position=SECOND_ROW, autoCommit=NO, batchType=PREPARED, generateKeys=NO, binary=FORCE, insertRewrite=true)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PGObjectGetTest.getAsPGobjectSubtype[binary = FORCE, sql = null::interval, type = class org.postgresql.util.PGInterval] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PGObjectGetTest.getAsPGobject[binary = REGULAR, sql = null::interval, type = class org.postgresql.util.PGInterval] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[98: batchTest(mode=FAIL_VIA_DUP_KEY, position=FIRST_ROW, autoCommit=YES, batchType=PREPARED, generateKeys=YES, binary=FORCE, insertRewrite=true)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.DatabaseMetaDataPropertiesTest.testCapabilities - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.DatabaseMetaDataPropertiesTest.testDriverVersioning - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PreparedStatementTest.testSetFloatInteger[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.CopyTest.testStatementCopyOut - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.CopyTest.testCopyInFromInputStream - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[3: batchTest(mode=NO_FAIL_SELECT, position=NONE, autoCommit=YES, batchType=SIMPLE, generateKeys=YES, binary=FORCE, insertRewrite=false)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[175: batchTest(mode=FAIL_VIA_DUP_KEY, position=ALMOST_LAST_ROW, autoCommit=NO, batchType=PREPARED_WITH_GENERATED, generateKeys=NO, binary=FORCE, insertRewrite=false)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchedInsertReWriteEnabledTest.test17000Binds[2: autoCommit=NO, binary=REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.CursorFetchTest.testResultSetFetchSizeFour[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.IntervalTest.testDaysHours - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PreparedStatementTest.testBatchWithPrepareThreshold5[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PreparedStatementTest.testSetObjectWithBigDecimal[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.DriverTest.testConnect - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ArrayTest.testSetNull[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaNonPrepared[returningInQuery = NO, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testCloseStatementClosesRS[returningInQuery = AB, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[174: batchTest(mode=FAIL_VIA_DUP_KEY, position=ALMOST_LAST_ROW, autoCommit=NO, batchType=PREPARED_WITH_GENERATED, generateKeys=NO, binary=FORCE, insertRewrite=true)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.CursorFetchTest.testSingleRowResultPositioning[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.MiscTest.testDatabaseSelectNullBug - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ColumnSanitiserDisabledTest.testTableColumnMixedNowFindFindMixedCaseColumn - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchedInsertReWriteEnabledTest.testMultiValues1bind[1: autoCommit=YES, binary=FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.CursorFetchTest.testInsert[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testEmptyRSWithoutReturning[returningInQuery = A, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PreparedStatementTest.testNaNLiteralsSimpleStatement[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ResultSetTest.testGetBadUuidBoolean - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testPSUpdate[returningInQuery = A, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.OuterJoinSyntaxTest.testOuterJoinSyntaxWithSingleJoinAndWithoutOj - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PreparedStatementTest.testSetObjectNumberFallbackWithAtomicLong[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ReplaceProcessingTest.run[input={fn database()}, expected=current_database()] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ResultSetTest.testStatementResultSetColumnMappingCache - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testStatementUpdateCount[returningInQuery = AB, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testDelete[returningInQuery = AB, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[153: batchTest(mode=FAIL_VIA_DUP_KEY, position=SECOND_ROW, autoCommit=YES, batchType=PREPARED_WITH_GENERATED, generateKeys=YES, binary=REGULAR, insertRewrite=false)] - unknown (unexpected)
		For a full summary look at the pgjdbc artifacts 
		
		An updated blocklist (pgjdbcBlockList22_1) is available in the artifacts' pgjdbc log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pgjdbc.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.pgjdbc [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4445360&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4445360&tab=artifacts#/pgjdbc) on master @ [a7449866f3229872d92e1de973e209838f07de35](https://github.com/cockroachdb/cockroach/commits/a7449866f3229872d92e1de973e209838f07de35):


```
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1159: quotes(style=$DEF$, src=z, quoted=--
		/* $ */$DEF$z$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1122: quotes(style=$DEF$, src=q, quoted=-- $
		$DEF$q$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[1425: quotes(style=$o‿o$, src=s, quoted=/*
		$
		*//* ? *//*{fn *//* now} */$o‿o$s$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaPreparedExecute[returningInQuery = STAR, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[831: quotes(style=$a$, src=r, quoted=--
		/* $ */$a$r$a$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1053: quotes(style=$DEF$, src=$o‿o$$o‿o$$o‿o$, quoted=/*
		$
		*//* ? *//*{fn *//* now} */$DEF$$o‿o$$o‿o$$o‿o$$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[87: quotes(style=', src={, quoted=--
		/* $ */'{')] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[123: quotes(style=', src=/*, quoted=--
		/* $ */'/*')] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[365: quotes(style=$$, src=select?select?select?, quoted=/*
		$
		*//* ? *//*{fn *//* now} */$$select?select?select?$$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[764: quotes(style=$a$, src=b, quoted=$a$b$a$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[956: quotes(style=$DEF$, src={{{, quoted=$DEF${{{$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[77: quotes(style=', src=}{}{, quoted=/*
		$
		*//* ? *//*{fn *//* now} */'}{}{')] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[377: quotes(style=$$, src=??select??select??select, quoted=/*
		$
		*//* ? *//*{fn *//* now} */$$??select??select??select$$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[1427: quotes(style=$o‿o$, src=s, quoted=--
		/* $ */$o‿o$s$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testGeneratedKeysCleared[returningInQuery = A, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.ResultSetTest.testHoldableResultSet - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[754: quotes(style=$a$, src=$o‿o$$o‿o$, quoted=-- $
		$a$$o‿o$$o‿o$$a$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[1125: quotes(style=$DEF$, src=r, quoted=/*
		$
		*//* ? *//*{fn *//* now} */$DEF$r$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testPSDelete[returningInQuery = NO, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[296: quotes(style=', src=x, quoted='x')] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[681: quotes(style=$a$, src=----, quoted=/*
		$
		*//* ? *//*{fn *//* now} */$a$----$a$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1109: quotes(style=$DEF$, src=n, quoted=/*
		$
		*//* ? *//*{fn *//* now} */$DEF$n$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[1334: quotes(style=$o‿o$, src=$a$$a$, quoted=-- $
		$o‿o$$a$$a$$o‿o$)] - unknown (unexpected)
		For a full summary look at the pgjdbc artifacts 
		
		An updated blocklist (pgjdbcBlockList22_1) is available in the artifacts' pgjdbc log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pgjdbc.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.pgjdbc [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4455314&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4455314&tab=artifacts#/pgjdbc) on master @ [978744dbb94b03497154547ce7fa97ea3be9a401](https://github.com/cockroachdb/cockroach/commits/978744dbb94b03497154547ce7fa97ea3be9a401):


```
		--- FAIL: org.postgresql.test.jdbc4.CharacterStreamTest.testKnownLongLength200Kb - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testUpdate[returningInQuery = A, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.XmlTest.testSetNull - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.breakDescribeOnFirstServerPreparedExecution[returningInQuery = AB, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testDelete[returningInQuery = STAR, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testGeneratedKeysCleared[returningInQuery = STAR, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.ClientInfoTest.testSetAppNameProps - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.Jdbc3SavepointTest.testRollbackMultipleTimes - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.BinaryStreamTest.testUnknownLength200Kb - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.WrapperTest.testStatementUnwrapStatement - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.BinaryStreamTest.testUnknownLengthEmpty - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PGObjectGetTest.getAsString[binary = FORCE, sql = null::interval, type = class org.postgresql.util.PGInterval] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testGeneratedKeys[returningInQuery = STAR, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.breakDescribeOnFirstServerPreparedExecution[returningInQuery = AB, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testCloseStatementClosesRS[returningInQuery = STAR, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testReturningWithTrailingSemicolon[returningInQuery = NO, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.Jdbc3SavepointTest.testRollingBackToSavepoints - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaPreparedExecute[returningInQuery = STAR, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.core.NativeQueryBindLengthTest.testBindLengthCalculation['$1$2...$9$10..$99$100'.length = 2*9+3*90+4 == 292] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.WrapperTest.testConnectionUnwrapPGConnection - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.JsonbTest.testJsonbNonPreparedStatement[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.jdbc41.CloseOnCompletionTest.testSingleResultSet - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.IntervalTest.testGetValueForSmallValueWithCommaAsDecimalSeparatorInDefaultLocale - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.IntervalTest.testMicroSecondsAreRoundedToNearest - unknown (unexpected)
		--- FAIL: org.postgresql.test.extensions.HStoreTest.testHStoreUsingPSSetObject4 - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.JsonbTest.testJsonbPreparedStatement[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testStatementUpdateCount[returningInQuery = STAR, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.XmlTest.testStAXRead - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.JsonbTest.jsonbArray[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testUpdate[returningInQuery = STAR, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testGeneratedKeysCleared[returningInQuery = STAR, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testCloseStatementClosesRS[returningInQuery = AB, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testEmptyRSWithoutReturning[returningInQuery = STAR, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaNonPrepared[returningInQuery = STAR, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.core.NativeQueryBindLengthTest.testBindLengthCalculation['$1...$1000'.length == 3,893] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.jdbc41.SchemaTest.testSchemaPath$User - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaPreparedExecute[returningInQuery = A, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.selectWithGeneratedKeysViaPreparedExecuteQuery[returningInQuery = A, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.Jdbc3SavepointTest.testContinueAfterError - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testMultipleRows[returningInQuery = A, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.jdbc41.AbortTest.abortWithNullExecutor - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.breakDescribeOnFirstServerPreparedExecution[returningInQuery = STAR, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.core.NativeQueryBindLengthTest.testBindLengthCalculation['$1...$10000'.length == 48,894] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.BinaryTest.testPreparedStatement_0 - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.BinaryTest.testPreparedStatement_negative1 - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testMultipleRows[returningInQuery = STAR, binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.jdbc.NoColumnMetadataIssue1613Test.shouldBeNoNPE - unknown (unexpected)
		For a full summary look at the pgjdbc artifacts 
		
		An updated blocklist (pgjdbcBlockList22_1) is available in the artifacts' pgjdbc log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pgjdbc.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.pgjdbc [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4464217&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4464217&tab=artifacts#/pgjdbc) on master @ [8e7248296a2b8b2a4d07dd82678c687a02d56c4f](https://github.com/cockroachdb/cockroach/commits/8e7248296a2b8b2a4d07dd82678c687a02d56c4f):


```
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[586: quotes(style=$a$, src={extract}{extract}, quoted=-- $
		$a${extract}{extract}$a$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[1243: quotes(style=$o‿o$, src=}{}{}{, quoted=--
		/* $ */$o‿o$}{}{}{$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1274: quotes(style=$o‿o$, src=----, quoted=-- $
		$o‿o$----$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[1325: quotes(style=$o‿o$, src=$$$$$$, quoted=/*
		$
		*//* ? *//*{fn *//* now} */$o‿o$$$$$$$$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1351: quotes(style=$o‿o$, src=$DEF$$DEF$$DEF$, quoted=--
		/* $ */$o‿o$$DEF$$DEF$$DEF$$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1244: quotes(style=$o‿o$, src={, quoted=$o‿o${$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1298: quotes(style=$o‿o$, src=*/*/, quoted=-- $
		$o‿o$*/*/$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[121: batchTest(mode=FAIL_VIA_DUP_KEY, position=ALMOST_LAST_ROW, autoCommit=YES, batchType=PREPARED, generateKeys=YES, binary=REGULAR, insertRewrite=false)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[252: quotes(style=', src=m, quoted='m')] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[380: quotes(style=$$, src=}{, quoted=$$}{$$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc3.GeneratedKeysTest.testWithInsertInsert[returningInQuery = NO, binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[722: quotes(style=$a$, src=''', quoted=-- $
		$a$'''$a$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.DatabaseMetaDataPropertiesTest.testIdentifiers - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[295: quotes(style=', src=w, quoted=--
		/* $ */'w')] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1194: quotes(style=$o‿o$, src={select}{select}{select}, quoted=-- $
		$o‿o${select}{select}{select}$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[56: quotes(style=', src=select?select?select?, quoted='select?select?select?')] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[884: quotes(style=$DEF$, src={extract}{extract}{extract}, quoted=$DEF${extract}{extract}{extract}$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1302: quotes(style=$o‿o$, src=*/*/*/, quoted=-- $
		$o‿o$*/*/*/$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1406: quotes(style=$o‿o$, src=n, quoted=-- $
		$o‿o$n$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchFailureTest.run[140: batchTest(mode=NO_FAIL_JUST_INSERTS, position=NONE, autoCommit=NO, batchType=PREPARED_WITH_GENERATED, generateKeys=NO, binary=REGULAR, insertRewrite=true)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[91: quotes(style=', src={{, quoted=--
		/* $ */'{{')] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[447: quotes(style=$$, src=*/*/, quoted=--
		/* $ */$$*/*/$$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[873: quotes(style=$DEF$, src={fn now}{fn now}{fn now}, quoted=/*
		$
		*//* ? *//*{fn *//* now} */$DEF${fn now}{fn now}{fn now}$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[964: quotes(style=$DEF$, src=}}, quoted=$DEF$}}$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[968: quotes(style=$DEF$, src=}}}, quoted=$DEF$}}}$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[1199: quotes(style=$o‿o$, src=?select, quoted=--
		/* $ */$o‿o$?select$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1307: quotes(style=$o‿o$, src=', quoted=--
		/* $ */$o‿o$'$o‿o$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.bindInTheMiddle[1040: quotes(style=$DEF$, src=$a$$a$$a$, quoted=$DEF$$a$$a$$a$$DEF$)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.QuotationTest.quotedString[336: quotes(style=$$, src={select}{select}, quoted=$${select}{select}$$)] - unknown (unexpected)
		For a full summary look at the pgjdbc artifacts 
		
		An updated blocklist (pgjdbcBlockList22_1) is available in the artifacts' pgjdbc log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pgjdbc.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.pgjdbc [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4467588&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4467588&tab=artifacts#/pgjdbc) on master @ [eee8ef080a95bc59eb3ddc2bebf623eb78e78db9](https://github.com/cockroachdb/cockroach/commits/eee8ef080a95bc59eb3ddc2bebf623eb78e78db9):


```
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testEmptyExecuteLargeBatchPreparedStatement[binary = REGULAR, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testGetLargeUpdateCountPreparedStatementSMALL[binary = FORCE, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeUpdatePreparedStatementSELECT[binary = REGULAR, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310Test.testBcTimestamp[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testSingleValuedUpsertUpdateBatch[binary = FORCE, reWriteBatchedInserts = YES] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testSingleValuedUpsertBatch[binary = FORCE, reWriteBatchedInserts = YES] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310Test.testGetLocalTime[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.SetObject310Test.testSetLocalTimeAndReadBack[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testUpsertDoNothingNoConflict[binary = REGULAR, reWriteBatchedInserts = NO] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testEmptyExecuteLargeBatchStatement[binary = FORCE, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.replication.ReplicationSlotTest.testNotAvailableCreateLogicalSlotWithoutSlotName - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testNullExecuteLargeBatchStatement[binary = FORCE, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.extensions.HStoreTest.testHStoreSelectNullValue - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeUpdateStatementSMALL[binary = FORCE, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.SetObject310Test.testSetLocalTimeWithType[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargePreparedStatementStatementSMALL[binary = FORCE, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.jdbc.NoColumnMetadataIssue1613Test.shouldBeNoNPE - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargePreparedStatementStatementSMALL[binary = FORCE, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310Test.testBcTimestamptz[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.replication.CopyBothResponseTest.initializationError - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testGetLargeUpdateCountStatementSELECT[binary = REGULAR, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testGetLargeUpdateCountPreparedStatementSELECT[binary = REGULAR, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.SetObject310Test.testSetOffsetDateTime[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.SetObject310Test.testSetLocalDateTimeBc[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testUpsertDoUpdateConflict[binary = FORCE, reWriteBatchedInserts = YES] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeBatchValuesInsertSMALL[binary = REGULAR, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeUpdateStatementSELECT[binary = REGULAR, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testGetLargeUpdateCountStatementSMALL[binary = REGULAR, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ArrayTest.testEmptyArray[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310InfinityTests.test[binary = FORCE, expr = infinity, pgType = date, klass = class java.time.LocalDate] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testEmptyExecuteLargeBatchPreparedStatement[binary = FORCE, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.util.ByteStreamWriterTest.testRuntimeExceptionPassedThroughAsIOException - unknown (unexpected)
		--- FAIL: org.postgresql.test.jre8.core.SocksProxyTest.testConnectWithSocksNonProxyHost - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ArrayTest.testDirectFieldString[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ArrayTest.testDirectFieldString[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeBatchStatementSMALL[binary = REGULAR, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeUpdatePreparedStatementSMALL[binary = REGULAR, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310Test.testProlepticCalendarTimestamptz[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ArrayTest.testNullString[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.replication.ReplicationSlotTest.testNotAvailableCreateLogicalSlotWithoutOutputPlugin - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.SetObject310Test.testSetLocalDateTimeBc[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeUpdatePreparedStatementSELECT[binary = REGULAR, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testSingleValuedUpsertUpdateConstantBatch[binary = FORCE, reWriteBatchedInserts = YES] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testUpsertDoNothingConflict[binary = FORCE, reWriteBatchedInserts = NO] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.PreparedStatementTest.testSetNumber - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ArrayTest.testNullFieldString[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeBatchStatementSMALL[binary = FORCE, insertRewrite = false] - unknown (unexpected)
		For a full summary look at the pgjdbc artifacts 
		
		An updated blocklist (pgjdbcBlockList22_1) is available in the artifacts' pgjdbc log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pgjdbc.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.pgjdbc [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4470933&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4470933&tab=artifacts#/pgjdbc) on master @ [9b45e990d5ddbbd6da13e8d3e312bb251fe0233a](https://github.com/cockroachdb/cockroach/commits/9b45e990d5ddbbd6da13e8d3e312bb251fe0233a):


```
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testMultiValuedUpsertBatch[binary = REGULAR, reWriteBatchedInserts = NO] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testUpsertDoNothingNoConflict[binary = FORCE, reWriteBatchedInserts = NO] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.BinaryStreamTest.testUnknownLength10Kb - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ResultSetTest.testGetOutOfBounds - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ResultSetTest.testClosedResult - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testUpsertDoUpdateNoConflict[binary = REGULAR, reWriteBatchedInserts = NO] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.WrapperTest.testConnectionUnwrapPGConnection - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.JsonbTest.jsonbArray[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testNullExecuteLargeBatchPreparedStatement[binary = FORCE, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.ConnectionValidTimeoutTest.testIsValidRespectsSmallerTimeout[networkTimeoutMillis=500, validationTimeoutSeconds=1, expectedMaxValidationTimeMillis=600] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ServerPreparedStmtTest.testEmptyResults - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeUpdateStatementSMALL[binary = FORCE, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ResultSetTest.testCaseInsensitiveFindColumn - unknown (unexpected)
		--- FAIL: org.postgresql.test.util.ByteStreamWriterTest.testRuntimeExceptionPassedThroughAsIOException - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ColumnSanitiserEnabledTest.testTableColumnUpperNowFindFindUpperCaseColumn - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.XmlTest.testDOMParse - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeBatchStatementSMALL[binary = REGULAR, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310Test.testBcTimestamptz[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ParameterStatusTest.parameterMapReadOnly - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.CharacterStreamTest.testKnownLongLength10Kb - unknown (unexpected)
		--- FAIL: org.postgresql.replication.CopyBothResponseTest.initializationError - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.BatchExecuteTest.testMixedBatch[binary = FORCE, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testGetLargeUpdateCountPreparedStatementSELECT[binary = FORCE, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ServerPreparedStmtTest.testSPSToggle - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.jdbc41.NetworkTimeoutTest.testSetNetworkTimeoutInvalid - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testUpsertDoUpdateConflict[binary = REGULAR, reWriteBatchedInserts = YES] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeUpdatePreparedStatementSELECT[binary = FORCE, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeUpdateStatementSMALL[binary = REGULAR, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testUpsertDoNothingConflict[binary = FORCE, reWriteBatchedInserts = NO] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ConcurrentStatementFetch.testFetchTwoStatements[3: fetch(autoCommit=YES, fetchSize=2, binaryMode=FORCE)] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargePreparedStatementStatementLoopSMALL[binary = REGULAR, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.IsValidTest.testIsValidShouldNotModifyTransactionStateInEmptyTransaction - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.WrapperTest.testStatementIsWrapperForPGStatement - unknown (unexpected)
		--- FAIL: org.postgresql.jdbc.NoColumnMetadataIssue1613Test.shouldBeNoNPE - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.ColumnSanitiserEnabledTest.testTableColumnMixedNowFindFindUpperCaseColumn - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.WrapperTest.testConnectionUnwrapConnection - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.GetObject310Test.testProlepticCalendarTimestamp[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.CursorFetchTest.testRowResultPositioningWithoutIsLast[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargeUpdateStatementSMALL[binary = FORCE, insertRewrite = true] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.CursorFetchTest.testScrollableAbsoluteFetch[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.IsValidTest.testIsValidShouldNotModifyTransactionStateOutsideTransaction - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc4.JsonbTest.jsonbArray[binary = FORCE] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.PGObjectGetTest.getAsString[binary = FORCE, sql = null::interval, type = class org.postgresql.util.PGInterval] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.UpsertTest.testSingleValuedUpsertUpdateBatch[binary = FORCE, reWriteBatchedInserts = NO] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc42.LargeCountJdbc42Test.testExecuteLargePreparedStatementStatementSMALL[binary = FORCE, insertRewrite = false] - unknown (unexpected)
		--- FAIL: org.postgresql.test.jdbc2.CursorFetchTest.testSingleRowResultPositioning[binary = REGULAR] - unknown (unexpected)
		--- FAIL: org.postgresql.test.util.ByteStreamWriterTest.testLengthGreaterThanContent - unknown (unexpected)
		For a full summary look at the pgjdbc artifacts 
		
		An updated blocklist (pgjdbcBlockList22_1) is available in the artifacts' pgjdbc log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pgjdbc.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/BUILD.bazel
+++ b/pkg/sql/BUILD.bazel
@@ -535,6 +535,7 @@ go_test(
         "show_trace_replica_test.go",
         "sort_test.go",
         "split_test.go",
+        "sql_cursor_test.go",
         "statement_mark_redaction_test.go",
         "table_ref_test.go",
         "table_test.go",

--- a/pkg/sql/sql_cursor.go
+++ b/pkg/sql/sql_cursor.go
@@ -30,10 +30,6 @@ import (
 // DeclareCursor implements the DECLARE statement.
 // See https://www.postgresql.org/docs/current/sql-declare.html for details.
 func (p *planner) DeclareCursor(ctx context.Context, s *tree.DeclareCursor) (planNode, error) {
-	if p.autoCommit {
-		return nil, pgerror.Newf(pgcode.NoActiveSQLTransaction, "DECLARE CURSOR can only be used in transaction blocks")
-	}
-
 	if s.Hold {
 		return nil, unimplemented.NewWithIssue(41412, "DECLARE CURSOR WITH HOLD")
 	}
@@ -44,61 +40,71 @@ func (p *planner) DeclareCursor(ctx context.Context, s *tree.DeclareCursor) (pla
 		return nil, unimplemented.NewWithIssue(41412, "DECLARE SCROLL CURSOR")
 	}
 
-	ie := p.ExecCfg().InternalExecutorFactory(ctx, p.SessionData())
-	cursorName := s.Name.String()
-	if cursor, _ := p.sqlCursors.getCursor(cursorName); cursor != nil {
-		return nil, pgerror.Newf(pgcode.DuplicateCursor, "cursor %q already exists", cursorName)
-	}
+	return &delayedNode{
+		name: s.String(),
+		constructor: func(ctx context.Context, p *planner) (_ planNode, _ error) {
+			if p.extendedEvalCtx.TxnImplicit {
+				return nil, pgerror.Newf(pgcode.NoActiveSQLTransaction, "DECLARE CURSOR can only be used in transaction blocks")
+			}
 
-	// Try to plan the cursor query to make sure that it's valid.
-	stmt := makeStatement(parser.Statement{AST: s.Select}, ClusterWideID{})
-	pt := planTop{}
-	pt.init(&stmt, &p.instrumentation)
-	opc := &p.optPlanningCtx
-	opc.p.stmt = stmt
-	opc.reset()
+			ie := p.ExecCfg().InternalExecutorFactory(ctx, p.SessionData())
+			cursorName := s.Name.String()
+			if cursor, _ := p.sqlCursors.getCursor(cursorName); cursor != nil {
+				return nil, pgerror.Newf(pgcode.DuplicateCursor, "cursor %q already exists", cursorName)
+			}
 
-	memo, err := opc.buildExecMemo(ctx)
-	if err != nil {
-		return nil, err
-	}
-	if err := opc.runExecBuilder(
-		&pt,
-		&stmt,
-		newExecFactory(p),
-		memo,
-		p.EvalContext(),
-		p.autoCommit,
-	); err != nil {
-		return nil, err
-	}
-	if pt.flags.IsSet(planFlagContainsMutation) {
-		// Cursors with mutations are invalid.
-		return nil, pgerror.Newf(pgcode.FeatureNotSupported,
-			"DECLARE CURSOR must not contain data-modifying statements in WITH")
-	}
+			// Try to plan the cursor query to make sure that it's valid.
+			stmt := makeStatement(parser.Statement{AST: s.Select}, ClusterWideID{})
+			pt := planTop{}
+			pt.init(&stmt, &p.instrumentation)
+			opc := &p.optPlanningCtx
+			opc.p.stmt = stmt
+			opc.reset()
 
-	statement := s.Select.String()
-	rows, err := ie.QueryIterator(ctx, "sql-cursor", p.txn, statement)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to DECLARE CURSOR")
-	}
-	inputState := p.txn.GetLeafTxnInputState(ctx)
-	cursor := &sqlCursor{
-		InternalRows: rows,
-		readSeqNum:   inputState.ReadSeqNum,
-		txn:          p.txn,
-		statement:    statement,
-		created:      timeutil.Now(),
-	}
-	if err := p.sqlCursors.addCursor(cursorName, cursor); err != nil {
-		// This case shouldn't happen because cursor names are scoped to a session,
-		// and sessions can't have more than one statement running at once. But
-		// let's be diligent and clean up if it somehow does happen anyway.
-		_ = cursor.Close()
-		return nil, err
-	}
-	return newZeroNode(nil /* columns */), nil
+			memo, err := opc.buildExecMemo(ctx)
+			if err != nil {
+				return nil, err
+			}
+			if err := opc.runExecBuilder(
+				&pt,
+				&stmt,
+				newExecFactory(p),
+				memo,
+				p.EvalContext(),
+				p.autoCommit,
+			); err != nil {
+				return nil, err
+			}
+			if pt.flags.IsSet(planFlagContainsMutation) {
+				// Cursors with mutations are invalid.
+				return nil, pgerror.Newf(pgcode.FeatureNotSupported,
+					"DECLARE CURSOR must not contain data-modifying statements in WITH")
+			}
+
+			statement := s.Select.String()
+			itCtx := context.Background()
+			rows, err := ie.QueryIterator(itCtx, "sql-cursor", p.txn, statement)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to DECLARE CURSOR")
+			}
+			inputState := p.txn.GetLeafTxnInputState(ctx)
+			cursor := &sqlCursor{
+				InternalRows: rows,
+				readSeqNum:   inputState.ReadSeqNum,
+				txn:          p.txn,
+				statement:    statement,
+				created:      timeutil.Now(),
+			}
+			if err := p.sqlCursors.addCursor(cursorName, cursor); err != nil {
+				// This case shouldn't happen because cursor names are scoped to a session,
+				// and sessions can't have more than one statement running at once. But
+				// let's be diligent and clean up if it somehow does happen anyway.
+				_ = cursor.Close()
+				return nil, err
+			}
+			return newZeroNode(nil /* columns */), nil
+		},
+	}, nil
 }
 
 var errBackwardScan = pgerror.Newf(pgcode.ObjectNotInPrerequisiteState, "cursor can only scan forward")
@@ -219,7 +225,12 @@ func (f fetchNode) Close(ctx context.Context) {
 // CloseCursor implements the FETCH statement.
 // See https://www.postgresql.org/docs/current/sql-close.html for details.
 func (p *planner) CloseCursor(ctx context.Context, n *tree.CloseCursor) (planNode, error) {
-	return newZeroNode(nil /* columns */), p.sqlCursors.closeCursor(n.Name.String())
+	return &delayedNode{
+		name: n.String(),
+		constructor: func(ctx context.Context, p *planner) (planNode, error) {
+			return newZeroNode(nil /* columns */), p.sqlCursors.closeCursor(n.Name.String())
+		},
+	}, nil
 }
 
 type sqlCursor struct {

--- a/pkg/sql/sql_cursor_test.go
+++ b/pkg/sql/sql_cursor_test.go
@@ -1,0 +1,123 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/stretchr/testify/require"
+)
+
+// Make sure that preparing a DECLARE doesn't cause problems.
+func TestPrepareDeclare(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{Insecure: true})
+	defer srv.Stopper().Stop(context.Background())
+	defer db.Close()
+
+	ctx := context.Background()
+	conn, err := db.Conn(ctx)
+	require.NoError(t, err)
+
+	t.Run("prepare_declare_raw_txn", func(t *testing.T) {
+		// Make sure that preparing a DECLARE defers errors until execution.
+		stmt, err := conn.PrepareContext(ctx, "DECLARE foo CURSOR FOR VALUES (1), (2)")
+		require.NoError(t, err)
+
+		_, err = stmt.Exec()
+		require.EqualError(t, err, "pq: DECLARE CURSOR can only be used in transaction blocks")
+
+		// Make sure that we can use our prepared statement from before to
+		// successfully execute a declare cursor within a transaction.
+		// We need to execute a raw BEGIN so that we can reuse our pre-prepared txn.
+		_, err = conn.ExecContext(ctx, "BEGIN TRANSACTION")
+		require.NoError(t, err)
+
+		_, err = stmt.Exec()
+		require.NoError(t, err)
+
+		stmt, err = conn.PrepareContext(ctx, "FETCH 2 foo")
+		require.NoError(t, err)
+		r, err := stmt.Query()
+		require.NoError(t, err)
+		var actual int
+		r.Next()
+		require.NoError(t, r.Scan(&actual))
+		require.Equal(t, 1, actual)
+		more := r.Next()
+		require.Equal(t, true, more)
+		require.NoError(t, r.Scan(&actual))
+		require.Equal(t, 2, actual)
+		more = r.Next()
+		require.Equal(t, false, more)
+		_, err = conn.ExecContext(ctx, "COMMIT")
+		require.NoError(t, err)
+	})
+
+	t.Run("prepare_declare_driver_txn", func(t *testing.T) {
+		// Make sure that we can use the driver-level txn support to do the same thing.
+		tx, err := conn.BeginTx(context.Background(), nil /* opts */)
+		require.NoError(t, err)
+		stmt, err := tx.Prepare("DECLARE foo CURSOR FOR VALUES (1), (2)")
+		require.NoError(t, err)
+		_, err = stmt.Exec()
+		require.NoError(t, err)
+
+		stmt, err = tx.Prepare("FETCH 2 foo")
+		require.NoError(t, err)
+		r, err := stmt.Query()
+		require.NoError(t, err)
+		var actual int
+		r.Next()
+		require.NoError(t, r.Scan(&actual))
+		require.Equal(t, 1, actual)
+		more := r.Next()
+		require.Equal(t, true, more)
+		require.NoError(t, r.Scan(&actual))
+		require.Equal(t, 2, actual)
+		more = r.Next()
+		require.Equal(t, false, more)
+
+		require.NoError(t, tx.Commit())
+	})
+
+	// Make sure that we can use the automatic prepare support (when sending
+	// placeholders) to do the same thing.
+	// TODO (jordan): This currently doesn't work, because we don't fully walk
+	// the tree typechecking expressions when sending a DECLARE through the
+	// optimizer. See issue #77067.
+	// When this limitation is lifted, the rest of this test should be uncommented.
+	t.Run("prepare_declare_placeholder", func(t *testing.T) {
+		_, err = conn.ExecContext(ctx, "DECLARE foo CURSOR FOR SELECT 1 WHERE $1", true)
+		require.Contains(t, err.Error(), "could not determine data type of placeholder")
+
+		skip.WithIssue(t, 77067, "placeholders in DECLARE not supported")
+
+		stmt, err := conn.PrepareContext(ctx, "FETCH 1 foo")
+		require.NoError(t, err)
+		r, err := stmt.Query()
+		require.NoError(t, err)
+		var actual int
+		r.Next()
+		require.NoError(t, r.Scan(&actual))
+		require.Equal(t, 1, actual)
+		more := r.Next()
+		require.Equal(t, false, more)
+	})
+}

